content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+---
+reviewers:
+- robscott
+title: Topology Aware Hints
+feature:
+  title: Topology Aware Hints
+  description: >
+    Enables topology aware routing with topology hints on EndpointSlices.
+
+content_type: concept
+weight: 10
+---
+
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+_Topology Aware Hints_ enable topology aware routing with topology hints
+included in EndpointSlices. This approach tries to keep traffic close to where
+it originated from. This can result in decreased costs and improved performance.
+
+<!-- body -->
+
+## Introduction
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+_Topology Aware Hints_ adds new functionality that can help keep traffic within
+the zone it originated from. This concept is commonly referred to as "Topology
+Aware Routing". The "Hints" here refer to a key implementation detail. The
+EndpointSlice controller adds hints to EndpointSlices to allocate each endpoint
+to a zone. Those hints are then consumed by components like kube-proxy as they
+configure how requests are routed.
+
+## Using Topology Aware Hints
+
+This feature can be enabled for individual Services by setting the
+`service.kubernetes.io/topology-aware-hints` annotation to `auto`. This tells
+the EndpointSlice controller to set topology hints if it is deemed safe.
+Importantly, this does not guarantee that hints will always be set.
+
+## How it Works
+
+The functionality enabling this feature is split into two components: The
+EndpointSlice controller and Kube-Proxy. This provides a high level overview of
+how each component implements this feature.
```

Comment:
nit: consider omitting this or moving it much later in the page. So long as the feature works how we describe it, most readers won't need to look up which component implements which bit.

(also, you might use a network plugin that bypasses kube-proxy entirely but still honors these hints).

Reply:
Makes sense, I can work on this.

Reply:
I'm not sure how to actually make this more generic. I think it's important to understand that there are 2 components in play here, and both have various safeguards that could result in this feature not doing anything even when it appears to be enabled. I can understand why we might want to use "Control Plane" instead of "EndpointSlice Controller", but I don't think we have a sufficiently generic and broadly understood term for the component kube-proxy represents.

Reply:
Ok, fair enough.